module_1/e2e/counter.spec.js

```diff
@@ -0,0 +1,17 @@
+describe('Counter E2E Test', () => {
+  it('increments and decrements counter', async () => {
+      await browser.url('/');
+
+      const counterValue = await $('[data-testid="counter-value"]');
+      const incrementBtn = await $('[data-testid="increment-btn"]');
+      const decrementBtn = await $('[data-testid="decrement-btn"]');
+
+      await expect(counterValue).toHaveText('5');
+
+      await incrementBtn.click();
+      await expect(counterValue).toHaveText('6');
+
+      await decrementBtn.click();
+      await expect(counterValue).toHaveText('5');
```

Comment:
I'd personally avoid coupling to the initial number “5” if the app’s default were to change. 
You could read the initial text, then assert +1/-1

 ```suggestion
       const initialValue = await counterValue.getText();

       await incrementBtn.click();
       await expect(counterValue).toHaveText(`${+initialValue + 1}`);

       await decrementBtn.click();
       await expect(counterValue).toHaveText(initialValue);
```